Close #9 feature/backend/get-issue-list-api

AimingBro opened the pull request:
## 공유할 사항
- query를 관리하는 디렉토리를 따로 만들었습니다. (src/model/query)
- 에러 응답이 대부분 비슷해서 일단 util에 에러 응답을 하는 함수를 하나 만들었습니다 (src/util/error-response)
- API 요청에 대해서 응답은 하고 있는데 아직 완벽하게 테스트 된것은 아니여서, 문제가 발생한다면 추후 수정하는 방식으로 진행해야 할 것 같습니다.
- 필터 데이터를 query 스트링으로 받아야 하는데, 처음에 json형식으로 처리를 해서 다시 수정했습니다. API문서도 수정하였습니다!

## 같이 논의할 사항
- 쿼리가 복잡해서 쿼리를 만드는 함수를 작성했는데, 코드가 많이 복잡합니다. 더 좋은 방법이 있을지 같이 봐주시면 좋을거 같습니다!
- dotenv가 두군데에서 import되는거 같은데, app.js에서 한번만 import하면 어떨까요?

AimingBro committed:
Feat: add error-response util function
-에러 종류에 따라 분기되는 에러 응답 함수

AimingBro committed:
Feat: write Controller for GET /issues

AimingBro committed:
Feat: write service code for GET /issues
    - label 정보 파싱 후 저장
    - assignee 정보 파싱 후 저장

AimingBro committed:
Feat: make Model for GET /issues API
    - query 디렉토리 생성
    - issueList query 생성 함수 작성

AimingBro committed:
Fix: change request data type json to query string
- get 요청에서 json 방식으로 데이터를 받도록 처리하여서 query 스트링으로
변경함
- 기존의 sql문에서 자잘한 오류를 발견하여 수정

AimingBro committed:
Chore: edit console.log

AimingBro committed:
Fix: fix error when any filter option not applied
아무 필터 옵션이 없을 때 발생한 에러 수정

AimingBro committed:
Chore: delete console.log

AimingBro committed:
Fix : fix get issues api response form

- success 추가
- 오타 수정
- issues -> data로 수정

bell-won commented on backend/src/controller/issue/issue.js:
```diff
@@ -1,10 +1,22 @@
-  const issues = await issueService.getIssue()
-  return res.json({ issues })
+const getIssues = async (req, res) => {
+  const filterValues = {
```
```javascript
    const issues = await issueService.getIssues(...req.query)
```
구조 분해 할당을 이용해도 좋을 것 같아요

AimingBro replied:
오 이런식으로 구조분해 할당을 하는 방법이 있는줄 몰랐는데 적용하도록 하겠습니다!

bell-won commented on backend/src/model/query/issue.js:
```diff
@@ -0,0 +1,50 @@
+  const conditions = []
+  if (filterValues.label !== undefined && filterValues.label.length > 0)
+    conditions.push('(IL.labelID IN(' + filterValues.label.join(',') + '))')
+  if (filterValues.isOpen !== undefined)
```
특정 부분이 반복되는데 아래와 같은 방법으로 수정 할 수도 있을 것 같아요
```javascript
const filtering = {
  isOpen: (isOpen) => `I.isOpeen=${isOpen}`,
  milestone: (milestone) => `I.milestoneId=${milestone}`
}
const filterKeys = Object.keys(filterValues)
for(const key of filterKeys)
  if ( filtering[key] ) conditions.push( filtering[filterValues.author]( filterValues[key]) )
```
코드는 줄어들 것 같은데 좋은 방법일지는 모르겠군요 허허..

AimingBro replied:
생각하지 못했던 방법인데, 좋은 방법인거 같습니다!! 이것도 적용하겠습니다.

bell-won reviewed (approved):
쿼리 작성하실 때 너무 어려웠을 것 같은데 고생 많으셨습니다.

hyex commented on backend/src/app.js:
```diff
@@ -1,4 +1,6 @@
 import express from 'express'
+import dotenv from 'dotenv'
```
`app.js`에서만 import 해도 다른 파일에서 사용할 수 있나요?

hyex commented on backend/src/util/error-response.js:
```diff
@@ -0,0 +1,18 @@
+export default (err, res) => {
```
이 파일이랑 response message, status code 같이 얘기해봐용

hyex reviewed (approved):
쿼리가 복잡하니 관련 로직도 꽤 복잡했을 것 같은데 너무 수고하셨습니다 :)

zunopark reviewed (approved):
수고하셨습니다!

AimingBro committed:
Refactor: apply some PR reviews to code

- DB label, assignee concat을 ;로 변경
- 프론트에서 배열로 label을 보내면 바로 배열로 변환되어 응답받을 수
있어서 배열로 변환하는 과정 삭제
- filtering 객체를 사용해서 함수 간략화

AimingBro committed:
Chore: delete console.log
